x-pack/plugins/spaces/server/lib/copy_to_spaces/copy_to_spaces.ts

```diff
@@ -4,42 +4,38 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import {
-  SavedObjectsClientContract,
-  SavedObjectsLegacyService,
-  SavedObject,
-} from 'src/core/server';
+import { SavedObject, KibanaRequest, CoreStart } from 'src/core/server';
 import { Readable } from 'stream';
-import { SavedObjectsClientProviderOptions } from 'src/core/server';
+import {
+  exportSavedObjectsToStream,
+  importSavedObjectsFromStream,
+} from '../../../../../../src/core/server';
 import { spaceIdToNamespace } from '../utils/namespace';
 import { CopyOptions, CopyResponse } from './types';
 import { getEligibleTypes } from './lib/get_eligible_types';
 import { createReadableStreamFromArray } from './lib/readable_stream_from_array';
 import { createEmptyFailureResponse } from './lib/create_empty_failure_response';
 import { readStreamToCompletion } from './lib/read_stream_to_completion';
+import { COPY_TO_SPACES_SAVED_OBJECTS_CLIENT_OPTS } from './lib/saved_objects_client_opts';
+
+export function copySavedObjectsToSpacesFactory(core: CoreStart, request: KibanaRequest) {
```

Comment:
I'd suggest you pass in only the dependencies you need instead of all of CoreStart. That way it doesn't really matter where `getImportExportObjectLimit` comes from (setup or start) since both are available in your route which uses this lib. It will also make it more explicit which dependencies this library function really depend on.